pkg/didcomm/common/service/destination_default.go

```diff
@@ -106,3 +110,65 @@ func CreateDestination(didDoc *diddoc.Doc) (*Destination, error) {
 		DIDDoc:            didDoc,
 	}, nil
 }
+
+func createLegacyDestination(didDoc *diddoc.Doc) (*Destination, error) {
+	didCommService, ok := diddoc.LookupService(didDoc, legacyDIDCommServiceType)
+	if !ok {
+		return nil, fmt.Errorf("create destination: missing DID doc service")
+	}
+
+	uri, err := didCommService.ServiceEndpoint.URI()
+	if uri == "" || err != nil {
+		return nil, fmt.Errorf("create destination: no service endpoint on didcomm service block in diddoc: %#v", didDoc)
+	}
+
+	if len(didCommService.RecipientKeys) == 0 {
+		return nil, fmt.Errorf("create destination: no recipient keys on didcomm service block in diddoc: %#v", didDoc)
+	}
+
+	if len(didCommService.Accept) == 0 {
+		didCommService.Accept = []string{legacyDIDCommServiceType}
+	}
+
+	return &Destination{
+		RecipientKeys:     convertAnyB58Keys(didCommService.RecipientKeys),
+		ServiceEndpoint:   model.NewDIDCommV1Endpoint(uri),
+		RoutingKeys:       convertAnyB58Keys(didCommService.RoutingKeys),
+		MediaTypeProfiles: didCommService.Accept,
+	}, nil
+}
+
+func convertAnyB58Keys(keys []string) []string {
```

Comment:
Move to some utils maybe? Jut a proposal, don't know if it fits Golang and af-go conventions